### ISSUE TITLE
withDefaults on a required prop should throw a warning

### ISSUE THREAD

**Issue by TheDutchCoder** [vuejs/language-tools#4708]
### Vue - Official extension or vue-tsc version

2.0.10

### VSCode version

1.92.2

### Vue version

3.4.31

### TypeScript version

5.5.4

### System Info

```shell
System:
    OS: macOS 14.6.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 99.94 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.7.0 - ~/.nvm/versions/node/v20.7.0/bin/node
    npm: 10.1.0 - ~/.nvm/versions/node/v20.7.0/bin/npm
  Browsers:
    Chrome: 127.0.6533.120
    Edge: 127.0.2651.105
    Safari: 17.6
```

### Steps to reproduce

Just declare props in a TS-based component, but make a prop required _and_ have a default.
There's a few issues with this:
1. Required props shouldn't have default value, so the IDE should warn against that
2. When _using_ the component, the IDE no longer warns you about missing required props
3. The browser _does_ warn you against missing required props

The bottom line is that the user isn't aware of the problem anywhere during development until the open the browser console (and even then it's not clear what the issue is: a default value on a required prop)

```ts
const props = withDefaults(defineProps<{
  content: string
}>(), {
  content: 'test'
})
```

### What is expected?

1. The IDE to either warn on the `content: string` line or on the `content: 'test'` line
2. The IDE to show missing required props during implementation

### What is actually happening?

Nothing, which doesn't make the user aware that a required prop is missing during implementation.

**Comment by TheDutchCoder** [comment]
Here's a [reproduction](https://play.vuejs.org/#eNrFUk1PwzAM/StRLtukqhWD0yhDfOwAB5iAYy9V63YZbRIl6Tap6n/HSdeuk8a0AxK32O/ZeX52TR+k9DcV0BkNdaKYNESDqSQpYp7fRdToiM4jzkoplCFPopRXJFOiJCM/cJEtHt0eMaZHjGnHCIP2A2yHgYFSFrEBjAgJU7ZxD3y2fwTDcLoPw6DlhcGgmnooMhE8Y7m/1oLjJLUlRzTBUlaAepeGCY6DzIhDLBYXhdi+upxRFXhdPllB8n0iv9Y7m4voUoEGtYGI9piJVQ6mhRefb7DDdw+WIq0KZJ8BP0CLorIaW9pjxVOUPeA5tS/OYcbzL73YGeC6G8oKtczG8SOKflvXfhv9IPfav3F1EW/QxX6dl91CTbbMrJ4hi6vCaI+kkDEOSyWkJs3+Auzm7baHzPGAGDpRuDwcx8yINgrHQzXz8cRrBffYaAVoAnZrJhefklQwr+uuB2maMLCpc6fUX+y/m0BIJsT9SVMQ+HNDDih2v8Cq5ge+0m40) on the Playground.

Comp2 has the squiggly lines and shows the missing required prop error (as well as the browser warning).
If you comment out Comp2, you'll see that Comp1 does **not** have the squiggly lines but the browser still shows the warning.

I wonder if this is a Vue compiler problem instead?

**Comment by KazariEX** [comment]
Using `vue@3.5` locally should show error.

**Comment by KazariEX** [comment]
And you might need to update Vue - Official to the latest version (2.0.28).

**Comment by TheDutchCoder** [comment]
> Using `vue@3.5` locally should show error.

It doesn't (see [Playground](https://play.vuejs.org/#eNrFUk1PwkAQ/SubvQAJtgp6wYrxg4MelKjHJqaWKSy2u5vdLZA0/e/ObmkpCRIOJt525r2ZffNmCnonpbfKgY5ooGPFpCEaTC5JGvH5TUiNDuk45CyTQhnyIDJ5QRIlMtLxfBfZ4s71HmOwxxjUjMCvPsB2GBjIZBoZwIiQYMZW7oHP6g+/HQ62YeBXvMBvVdM+iowFT9jcW2rBcZLCkkMaYylLQb1KwwTHQUbEIRaL0lSsn13OqBz6dT5eQPx9IL/UG5sL6VSBBrWCkDaYidQcTAVP3l9gg+8GzMQsT5F9BHwDLdLcaqxo9zmfoewWz6l9cg4zPv/Qk40BruuhrFDLLB0/pOi3de230Xdyh96lqwt5iS426zztFgqyZmbxCEmUp0b3yQwSxmGqhNSk3F6A3bzddpvZbREDJwqXh+OYEdFG4XioZtzt9SvBDdZZAJqA3creyackFYyLou5ByjLwberYKTUX++8mEJIIcXvQFAT+3JAdit1PsupzBcreIDo19K6887MvMJE3pOUPVfZ1pQ==))

**Comment by KazariEX** [comment]
Debugging on the playground is difficult, please try it locally and ensure that `tsconfig.json` contains the following options:

```json
"vueCompilerOptions": {
	"target": 3.5
}
```

**Comment by KazariEX** [comment]
Weird, it works on my machine.

![image](https://github.com/user-attachments/assets/f0604d9e-dc78-4768-b51e-7b2c5f1fd1ac)

![image](https://github.com/user-attachments/assets/79d5dd92-61b6-41a7-9328-571bf83478a1)

Check if the declaration file points to the correct version? Witch should include `MakeDefaultsOptional` that controls this behavior.

**Comment by TheDutchCoder** [comment]
I'm not sure what I'm supposed to check (and where).

Side note: the VSCode extension only has version 2.0.10 available, no other versions are released as far as VSCode can tell, maybe that's the issue? (An older version of the tools?)
![image](https://github.com/user-attachments/assets/fee548a2-004d-4065-bea6-57221deaca51)
